tests/appservice/test_scheduler.py

```diff
@@ -49,11 +49,10 @@ def test_single_service_up_txn_sent(self):
         txn = Mock(id=txn_id, service=service, events=events)
 
         # mock methods
-        self.store.get_appservice_state = Mock(
-            return_value=defer.succeed(ApplicationServiceState.UP)
-        )
-        txn.send = Mock(return_value=make_awaitable(True))
-        self.store.create_appservice_txn = Mock(return_value=defer.succeed(txn))
+        self.store.get_appservice_state = simple_async_mock(ApplicationServiceState.UP)
+        txn.send = simple_async_mock(True)
+        txn.complete = simple_async_mock(True)
+        self.store.create_appservice_txn = simple_async_mock(txn)
```

Comment:
is there any change in behaviour here, or is this just a cleaner way of doing it?

Reply:
There's one bugfix in here, which is also handling `txn.complete`, which previously ended up as just a `Mock` which can't be awaited. The rest is just refactoring to be a bit cleaner. (Sorry for not separating those!)

While I was in here I also converted `defer.succeed` since the mocked functions don't return `Deferreds` anymore. Note that we (annoyingly) have two ways of making async mocks at the moment (`simple_async_mock` and `make_awaitable`), essentially you can switch `defer.succeed` -> `make_awaitable` OR you can switch `Mock(return_value=defer.succeed(...))` -> `simple_async_mock(...)`. Sad times.

Reply:
got it. Thanks!